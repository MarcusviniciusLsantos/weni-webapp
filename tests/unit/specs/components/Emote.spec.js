import { shallowMount } from '@vue/test-utils';
import Emote from '@/components/Emote.vue';

describe('Emote.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Emote, {
      mocks: {
        $t: () => '',
      },
    });
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('is smile the initial emote', () => {
    const SMILE = 3;
    expect(wrapper.vm.$data.current).toBe(SMILE);
  });

  it('does current data change when user pass the mouse enter', () => {
    const oldCurrent = wrapper.vm.$data.current;

    wrapper.find('span').trigger('mouseenter');
    expect(wrapper.vm.$data.current).not.toBe(oldCurrent);
  });

  it('does lastEmote value from localStorage change when user change the emote', async () => {
    const oldLastEmote = localStorage.getItem('lastEmote');
    await wrapper.find('span').trigger('mouseenter');
    expect(localStorage.getItem('lastEmote')).not.toBe(oldLastEmote);
  });
});
