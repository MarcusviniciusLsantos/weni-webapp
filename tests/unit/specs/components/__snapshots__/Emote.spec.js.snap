// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emote.vue renders correctly 1`] = `<span class="weni-emote"><img src=""></span>`;
